src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java

```diff
@@ -193,6 +198,45 @@ protected void setExpandedState(TreePath path, boolean state) {
                 }
             }
         };
+
+        /* Custom renderer to make the text bold when hovering on leaf nodes (entities) */
+        DefaultTreeCellRenderer renderer = new DefaultTreeCellRenderer() {
+            private Font originalFont;
+            private Font boldFont;
+
+            @Override
+            public Component getTreeCellRendererComponent(JTree tree, Object value, boolean selected, boolean expanded, boolean leaf, int row, boolean hasFocus) {
+                JLabel label = (JLabel) super.getTreeCellRendererComponent(tree, value, selected, expanded, leaf, row, hasFocus);
+
+                /* Apply custom styling only to leaf nodes (entities) */
+                if (leaf) {
+                    if (originalFont == null) {
+                        originalFont = label.getFont();
+                        boldFont = originalFont.deriveFont(Font.BOLD);
+                    }
+
+                    if (tree.getSelectionPath() != null && tree.getSelectionPath().equals(tree.getPathForRow(row))) {
+                        label.setFont(boldFont);
+                    } else {
+                        label.setFont(originalFont);
+                    }
+
+                    if (selected) {
+                        /* Avoid background color change */
+                        label.setBackground(tree.getBackground());
+                    }
+
+                    /* Ensure label preferred size can accommodate the full text */
+                    label.setPreferredSize(new Dimension(label.getPreferredSize().width + 20, label.getPreferredSize().height));
+                } else {
+                    /* Reset to original font for non-leaf nodes (groups) */
+                    label.setFont(originalFont);
+                }
+
+                return label;
+            }
+        };
+
```

Comment:
Does this actually do anything? I haven't seen any case where the font becomes bold and, in your screenshots, the font looks the same for both nodes and leaves whether they are selected or not.

Reply:
Yes! Do set the current hightlight I had to all of this, otherwise the text would get cut at the end because of highlight. I don't know if you could improve this though.

Reply:
Not sure I see the difference, maybe an OS-dependent kind of thing? Can you attach a screenshot or something with the difference, please?

Reply:
I just removed that whole code block and is still working, but I remember I struggled a bit on this. I don't know I got confused. I will push without this part then. It's working:

![image](https://github.com/user-attachments/assets/f3c8a762-b437-45d8-a907-666c89d5069a)